lib/active_shipping/carriers/usps_returns.rb

```diff
@@ -0,0 +1,99 @@
+require 'cgi'
+
+module ActiveShipping
+
+  class USPSReturns < Carrier
+
+    self.retry_safe = true
+
+    cattr_reader :name
+    @@name = "USPS Returns"
+
+    LIVE_DOMAIN = 'returns.usps.com'
+    LIVE_RESOURCE = 'Services/ExternalCreateReturnLabel.svc/ExternalCreateReturnLabel'
+
+    TEST_DOMAINS = { #indexed by security; e.g. TEST_DOMAINS[USE_SSL[:rates]]
+      true => 'returns.usps.com',
+      false => 'returns.usps.com'
+    }
+
+    TEST_RESOURCE = 'Services/ExternalCreateReturnLabel.svc/ExternalCreateReturnLabel'
+
+    API_CODES = {
+      :external_return_label_request => 'externalReturnLabelRequest'
+    }
+
+    USE_SSL = {
+      :external_return_label_request => true
+    }
+
+    def requirements
+      []
+    end
+
+    def external_return_label_request(label, options = {})
+      response = commit(:external_return_label_request, URI.encode(label.to_xml), (options[:test] || false))
```

Comment:
As @kmcphillips was kind of getting at but missed this part. The URI encoding should really be done in the `request_url` method, not here.